### ISSUE TITLE
GRPO——metric_for_best_model

### ISSUE THREAD

**Issue by After-lifes** [modelscope/ms-swift#9733]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new question or discussion topic. / 我已经搜索过现有的 issues，确认这是一个新的问题与讨论。

### Question Description / 问题描述

<img width="731" height="53" alt="Image" src="https://github.com/user-attachments/assets/0940dc64-96d1-4744-acb1-34c8f23bca3c" />
需要额外进行手动设置吗？

### How to Reproduce / 如何复现
ms-swift==4.0.4,transformers==4.57.6
在未手动设置metric_for_best_model时，使用swift进行GRPO训练时，生成的args文档中metric_for_best_model仍为“loss”而不是“reward”。既然GRPO是在reward激励下进行训练的，为什么仍使用loss作为衡量模型性能好坏的标准呢？

### Additional Information / 补充信息
另外，early_stop_interval是根据什么早停的，metric_for_best_model的设置能否对early_stop_interval起作用？

**Comment by hjh0119** [comment]
https://github.com/modelscope/ms-swift/blob/3d61b9318b27fdd5659e530cd36db7f4ce740fd7/swift/arguments/rlhf_args.py#L474-L476

**Comment by After-lifes** [comment]
[https://github.com/modelscope/ms-swift/blob/main/swift/arguments/sft_args.py](url)
line 421-425
if self.metric_for_best_model is None:
    self.metric_for_best_model = (
        "rouge-l" if self.predict_with_generate else "loss"
    )

**Comment by After-lifes** [comment]
初始化时，metric_for_best_model似乎直接拿到了“loss”，导致self.rlhf_type == 'grpo' and self.metric_for_best_model is None并不会成立。




> [ms-swift/swift/arguments/rlhf_args.py](https://github.com/modelscope/ms-swift/blob/3d61b9318b27fdd5659e530cd36db7f4ce740fd7/swift/arguments/rlhf_args.py#L474-L476)
> 
> Lines 474 to 476 in [3d61b93](/modelscope/ms-swift/commit/3d61b9318b27fdd5659e530cd36db7f4ce740fd7)
> 
>  def _init_metric_for_best_model(self): 
>      if self.rlhf_type == 'grpo' and self.metric_for_best_model is None: 
>          self.metric_for_best_model = 'reward'

**Comment by hjh0119** [comment]
> 初始化时，metric_for_best_model似乎直接拿到了“loss”

我这测试是正常的，提供一下复现脚本吧

**Comment by After-lifes** [comment]
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
 
MAX_PIXELS=501760 \
CUDA_VISIBLE_DEVICES=0,1 \
NPROC_PER_NODE=2 \
swift rlhf \
    --rlhf_type grpo \
    --model "${MODEL_PATH}" \
    --train_type lora \
    --freeze_vit true \
    --freeze_aligner true \
    --lora_rank 8 \
    --lora_alpha 32 \
    --torch_dtype bfloat16 \
    --load_from_cache_file false \
    --num_train_epochs 1 \
    --max_length 8192 \
    --per_device_train_batch_size 3 \
    --per_device_eval_batch_size 3 \
    --gradient_accumulation_steps 1 \
    --save_steps 30 \
    --eval_steps 30 \
    --learning_rate 6e-5 \
    --save_total_limit 5 \
    --save_only_model true \
    --repetition_penalty 1.1 \
    --logging_steps 1 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 2 \
    --dataset_num_proc 2 \
    --num_generations 6 \
    --temperature 0.9 \
    --beta 0.08 \
    --log_completions true \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.35 \
    --vllm_tensor_parallel_size 2 \
    --max_completion_length 3072 \
    --deepspeed zero3 \
    --remove_unused_columns false \
    --split_dataset_ratio 0.01 \
    --adapters "${SFT_ADAPTER}" \
    --ref_adapters "${SFT_ADAPTER}" \
    --output_dir "${OUTPUT_DIR}" \
    --dataset "${DATA_FILE}" \
    --reward_funcs ******** \
    --external_plugins "${PLUGIN_FILE}" \
    --sleep_level 1 \
    --offload_optimizer true \
    --offload_model true \
    --early_stop_interval 10 \
    --system "${SYS_PROMPT}" \
    --attn_impl flash_attn

**Comment by hjh0119** [comment]
> ms-swift==4.0.4

升级swift

**Comment by hjh0119** [comment]
#9336

**Comment by After-lifes** [comment]
> > ms-swift==4.0.4
> 
> 升级swift

必须升级吗，能通过手动设置来避免这个问题吗？

**Comment by hjh0119** [comment]
--metric_for_best_model reward 不行吗

**Comment by After-lifes** [comment]
> --metric_for_best_model reward 不行吗

可以的，设置过后生成的args文档中metric_for_best_model就是“reward”。但是我不想改动环境，除了这个参数，还要设置别的参数吗？

**Comment by hjh0119** [comment]
应该是不需要的，有具体的bug可以再提一个issue

**Comment by After-lifes** [comment]
> 应该是不需要的，有具体的bug可以再提一个issue

OK, thanks
